### PR TITLE
Merlins login

### PR DESCRIPTION
**Pull request by Magitron**
Login page for Merlin's Potions.

## Todo
- [x] style login 
- [x] remove temporary util for form encoding in `app/utils/utils.js`

## How to test-drive this PR
- Run Merlins
- Make an account on desktop if you haven't already
- Go to login and sign in

### CODE REVIEW

File: app/containers/login/partials/form.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        <form
+            onSubmit={handleSubmit((values) => {
+                return new Promise((resolve, reject) => {
+                    submitForm(values, resolve, reject)
+                })
+            })}
+            noValidate={true}
+        >
+            {error && <div className="u-margin-bottom-md u-color-error">{error}</div>}
+            <FormFields items={items} />
```
Comment (jeffkamo) [inline app/containers/login/partials/form.js]: @DanielBrookRoberge are we still recommending this be used? Are we going to move away from generating components in this manner vs. being declarative? i.e...

```
<Form>
    <FieldRow><Field {...} /></FieldRow>
    <FieldRow><Field {...} /></FieldRow>
</Form>
```

File: app/styles/lib/_color.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+// Base Color
+// ---
+//
+// These are just a helper mixins to aid in creating all the below mixins
+
+
+// Neutral Colors
+// ---
+@mixin bg-color-neutral-20(
```
Comment (jeffkamo) [inline app/styles/lib/_color.scss]: I suppose it's okay if you want to flesh out these helpers here, but maybe add a `@TODO` to that this file needs to (and should) be migrated over to Spline pronto

File: app/styles/utilities/_box-shadow.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Box Shadow
+// ===
+
+.u-box-shadow {
+    box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.3);
```
Comment (jeffkamo) [inline app/styles/utilities/_box-shadow.scss]: Good call, this is a fairly common utiliy. Please add a `@TODO` note to remind me to migrate this to Spline at a later date.

File: app/styles/utilities/_color.scss [hunk 4]
```diff
@@ -44,6 +44,10 @@
 // Background Colors
 // ---
 
+.u-bg-color-neutral-20 {
+    @include bg-color-neutral-20(true);
```
Comment (jeffkamo) [inline app/styles/utilities/_color.scss]: Same as the other utility, please add a `@TODO` reminder so I come back and migrate this to spline at a later date.

File: app/styles/utilities/_text.scss [hunk 5]
```diff
@@ -108,3 +108,10 @@
 .u-text-small {
     font-size: 12px !important; // @TODO: Add to spline
 }
+
+// Font Family
+// ---
+
+.u-text-font-family {
+    font-family: $font-family;
```
Comment (jeffkamo) [inline app/styles/utilities/_text.scss]: please add ` // @TODO: Add to spline`

File: app/containers/login/partials/form.js (hunk 1 above)
Comment (DanielBrookRoberge) [inline app/containers/login/partials/form.js]: I think we're aiming for the latter; @mikenikles would know for sure.

Comment (jeffkamo) [inline app/containers/login/partials/form.js]: @Magitron it looks like the "Forgot your password?" and "What's this?" links are missing.

Comment (jeffkamo) [inline app/containers/login/partials/form.js]: Also the `*` charcters at the end of the label, and the `* Required Fields` message are missing too

Comment (mikenikles) [inline app/containers/login/partials/form.js]: Yes we're looking to use the declarative approach where possible.

Comment (jeffkamo) [inline app/containers/login/partials/form.js]: @Magitron Also missing is the "!" icons from the caption.

Make sure the error caption and field borders inherit the correct error color. Right now they are set to a value of `red` which is coming from the SDK base styles, but they should use the local `$error-color`

File: app/containers/login/partials/form.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    return (
+        <form
+            onSubmit={handleSubmit((values) => {
+                return new Promise((resolve, reject) => {
+                    submitForm(values, resolve, reject)
+                })
+            })}
+            noValidate={true}
+        >
+            {error && <div className="u-margin-bottom-md u-color-error">{error}</div>}
```
Comment (Magitron) [inline app/containers/login/partials/form.js]: @guisers

File: app/containers/login/partials/form.js (hunk 1 above)
Comment (Magitron) [inline app/containers/login/partials/form.js]: @guisers

File: app/components/field/_theme.scss [hunk 7]
```diff
@@ -1,17 +1,33 @@
...
+    &.c--error {
+        .c-field__label {
+            color: $font-color; // 1
+        }
+    }
+}
+
+.c-field + .c-field {
+    margin-top: $unit*2;
+}
```
Comment (Magitron) [inline app/components/field/_theme.scss]: Makes sense, except that we aren't using field rows in this case. @guisers

Comment (guisers) [inline app/components/field/_theme.scss]: We aren't explicitly using `FieldRow`. I'm actually using our `FormFields` component which blackboxes all the field stuff. I just looked at the source code though and this component _does_ create `FieldRow` for me so it should be there

File: app/containers/login/partials/form.js (hunk 6 above)
Comment (guisers) [inline app/containers/login/partials/form.js]: @jeffkamo that's so weird, this is what it does on mine:
image

File: app/components/field/_theme.scss (hunk 7 above)
Comment (guisers) [inline app/components/field/_theme.scss]: Oh I see this is talked about later in the PR. I was reading this chronologically. Will refactor this the declarative way now 😅

File: app/containers/login/partials/form.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            <FormFields items={items} />
+            <button className="c-button c--primary u-width-full u-margin-top-lg" type="submit" disabled={submitting}>{submitText}</button>
```
Comment (guisers) [inline app/containers/login/partials/form.js]: @Magitron wanna do this w/css?

File: app/containers/login/partials/form.js (hunk 1 above)
Comment (guisers) [inline app/containers/login/partials/form.js]: Sounds good. Might be worth adding a note in the docs if this is the case

File: app/components/field/_theme.scss (hunk 7 above)
Comment (jeffkamo) [inline app/components/field/_theme.scss]: Thanks @guisers!

@Magitron now that the FieldRow component is being used, this `field + field` style can probably be removed.

File: app/containers/login/partials/form.js (hunk 8 above)
Comment (jeffkamo) [inline app/containers/login/partials/form.js]: there's a utility class you can use: `u-text-uppercase` I believe is the one

**Comment by guisers** [comment]
@jeffkamo Could you please check if the PR is meeting your requested changes? I _think_ we have addressed everything. :)

**Pull request merged**